### ISSUE TITLE
Running snippets before plugin so the plugin can modify the content

### ISSUE THREAD

**Issue by kiyoon** [mkdocs/mkdocs#3890]
Hi, I noticed that my page with a simple snippet like

```
--8<-- "README.md"
```

wasn't rendered correctly because I was using a plugin. I opened the issue in the plugin project, but apparently, they had no control (https://github.com/sondregronas/mkdocs-callouts/issues/24).

Is this something that could be fixed? Thank you!

**Comment by facelessuser** [comment]
The Snippets extension is a Python Markdown extension and is run as a preprocessor in Python Markdown. The MkDocs `on_page_markdown` event, which `mkdocs-callouts` seems to use, is run before Python Markdown is ever run (including Snippets).

To be honest, a MkDocs plugin trying to do Markdown work before the Markdown parser gets run is kind of weird. It would make more sense to have a Markdown extension processing callouts instead. Such an approach would be compatible with Python Markdown and Snippets.

So, in short, Snippets is not compatible with `mkdocs-callouts`. I don't personally agree with the approach of `mkdocs-callouts`, but it gets the job done so long as you don't care about other Markdown side effects due to its approach.

**Comment by pawamoy** [comment]
I agree with @facelessuser here. Nothing to be done by MkDocs or PyMDown Extensions, unfortunately, so I'll close this.

**Comment by kiyoon** [comment]
Thanks for the clear explanation. I had no idea, but now I do understand how it works!

I found a very simple markdown extension and using that works like expected. Thanks again!

https://github.com/dahlia/markdown-gfm-admonition
